integration/store_gateway_limits_hit_test.go

```diff
@@ -0,0 +1,202 @@
+	waitUntilShippedToStorage(t, mimirServices, 1)
+
+	cfg := grpcclient.Config{}
+	flagext.DefaultValues(&cfg)
+	reg := prometheus.NewPedanticRegistry()
+	factory := storegateway.NewStoreGatewayClientFactory(cfg, reg)
+	storeGatewayClient, err := factory(mimirServices[storeGatewayTag].GRPCEndpoint())
+	require.NoError(t, err)
+
+	ctx := prepareGRPCContext()
+	req := &storepb.SeriesRequest{
+		MinTime: timeStamps[series1].Unix() * 1000,
+		MaxTime: timeStamps[series4].Add(1*time.Hour).Unix() * 1000,
+		Matchers: []storepb.LabelMatcher{
+			{Type: storepb.LabelMatcher_RE, Name: labels.MetricName, Value: "series_.+"},
+		},
+	}
+	stream, err := storeGatewayClient.(*storegateway.ClientImpl).Series(ctx, req)
+	require.NoError(t, err)
+	require.NotNil(t, stream)
+
+	res, err := stream.Recv()
+	require.Nil(t, res)
+	require.NotNil(t, err)
+	st, ok := status.FromError(errors.Cause(err))
+	require.True(t, ok)
+	require.Equal(t, uint32(http.StatusUnprocessableEntity), uint32(st.Code()))
+}
+
+func TestStoreGateway_QuerierLimitHit(t *testing.T) {
+	scenario, mimirServices, client := createClient(t, map[string]string{"-querier.max-fetched-series-per-query": "3"})
+	defer scenario.Close()
+
+	timeStamps12, timeSeries12 := createTimeSeries(t, []string{series1, series2})
+	pushTimeSeries(t, client, timeSeries12[series1])
+	pushTimeSeries(t, client, timeSeries12[series2])
+
+	// ensures that data will be queried from store-gateway, and not from ingester: at this stage 1 block of data should be stored
+	waitUntilShippedToStorage(t, mimirServices, 1)
+
+	// Verify we can successfully read the data we have just pushed
+	rangeResultResponse, _, err := client.QueryRangeRaw("{__name__=~\"series_.+\"}", timeStamps12[series1], timeStamps12[series2].Add(1*time.Hour), time.Second)
+	require.NoError(t, err)
+	require.Equal(t, http.StatusOK, rangeResultResponse.StatusCode)
+
+	timeStamps34, timeSeries34 := createTimeSeries(t, []string{series3, series4})
+	pushTimeSeries(t, client, timeSeries34[series3])
+	pushTimeSeries(t, client, timeSeries34[series4])
+
+	// ensures that data will be queried from store-gateway, and not from ingester: at this stage 2 blocks of data should be stored
+	waitUntilShippedToStorage(t, mimirServices, 2)
+
+	// Verify we cannot read the data we just pushed because the limit is hit, and the status code 422 is returned
+	rangeResultResponse, rangeResultBody, err := client.QueryRangeRaw("{__name__=~\"series_.+\"}", timeStamps12[series1], timeStamps34[series4].Add(1*time.Hour), time.Second)
+	require.NoError(t, err)
+	require.Equal(t, http.StatusUnprocessableEntity, rangeResultResponse.StatusCode)
+	require.JSONEq(t, `{"status":"error","errorType":"execution","error":"expanding series: the query exceeded the maximum number of series (limit: 3 series) (err-mimir-max-series-per-query). To adjust the related per-tenant limit, configure -querier.max-fetched-series-per-query, or contact your service administrator."}`, string(rangeResultBody))
```

Comment:
Instead of requiring that the JSON match exactly, seems sufficient to just make sure it contains the expected global error name (`err-mimir-max-series-per-query`).